hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/components/HawkbitUIErrorHandler.java

```diff
@@ -38,10 +42,21 @@ public void error(final ErrorEvent event) {
 
         final Optional<Page> originError = getPageOriginError(event);
 
+        final HawkbitErrorNotificationMessage message = buildNotification(getRootExceptionFrom(event));
+
         if (originError.isPresent()) {
-            final HawkbitErrorNotificationMessage message = buildNotification(getRootExceptionFrom(event));
+            final Connector connector = ((ConnectorErrorEvent) event).getConnector();
```

Comment:
Well, I see. But I'd not rely on that knowlege. This dependency is hard to figure out and if you'll only look at class level `ConnectorErrorEvent` comes never into play. `findAbstractComponent(...)` is returning a `Component`. JavaDoc is also not pointing out that contract:

```
final Component errorOrigin = findAbstractComponent(event);

if (errorOrigin != null && errorOrigin.getUI() != null) {
     return Optional.fromNullable(errorOrigin.getUI().getPage());
}

return Optional.absent();  
```